src/main/java/core/basesyntax/strategy/impl/SupplyOperation.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.Operation;
+import core.basesyntax.strategy.OperationException;
+
+public class SupplyOperation implements Operation {
+    private final StorageDao storageDao;
+
+    public SupplyOperation(StorageDao storageDao) {
+        this.storageDao = storageDao;
+    }
+
+    @Override
+    public void proceed(FruitTransaction fruitTransaction) {
+        if (storageDao.get(fruitTransaction.getFruitName()) == null) {
+            throw new OperationException("Fruit " + fruitTransaction.getFruitName()
+                    + " doesn't exist");
```

Comment:
The supply operation should be able to add new fruits to the storage. Throwing an exception if the fruit doesn't exist is not needed here. Instead, you should check if the fruit exists in the storage and if not, create a new record with the supplied quantity.